### PR TITLE
[multistage] Adding v2 map function supports

### PR DESCRIPTION
**Pull request by xiangfu0**
- Support mapValue function
- Add MultiStageEngineMapTypeClusterIntegrationTest

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java [hunk 1]
```diff
@@ -125,7 +125,10 @@ public enum TransformFunctionType {
   ARRAYMAX("arrayMax"),
   ARRAYSUM("arraySum"),
   VALUEIN("valueIn"),
-  MAPVALUE("mapValue"),
+  MAPVALUE("mapValue", ReturnTypes.cascade(opBinding ->
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:128]: ARG2 is an array of X, X={INT, STRING, ...}, the return table should be this X.

Comment (walterddr) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:128]: you meant `X = {INT | STRING | ...}` correct?

**Review by walterddr — APPROVED** [review]
lgtm.

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:128]: yes

**Pull request merged**